.github/workflows/check-links.yaml

```diff
@@ -0,0 +1,51 @@
+name: check-links
+on:
+  push:
+    branches: [main]
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
```

Comment:
Why is this needed?

Reply:
Thank you for the review @tylerbenson  

The `cancel-in-progress: true` was included to ensure that any currently running workflows from the same PR would be canceled when a new commit is pushed, thus avoiding redundant workflow runs. This helps streamline the CI process, but I realize that this may not be necessary in this context. 

I can remove it if it's not aligned with the project's workflow policies.

Reply:
Assuming this job runs quickly, which I assume it does, I don't think that's necessary.  Go ahead and remove.

Reply:
"Thank you for the feedback, removed now.